scripts/upgrade_cfme.py

```diff
@@ -0,0 +1,216 @@
+#!/bin/python
+
+import argparse
+import yaml
+import time
+import threading
+import requests
+from datetime import datetime
+from utils.appliance import IPAppliance
+import logging
+
+# setting script logging object
+# logging all the event to script and progress messages to screen
+formatter = "[%(asctime)s][%(levelname)s][%(funcName)s] - %(message)s"
+log_name = "cfme-upgrade-{date}.log".format(date=datetime.now().strftime("%y:%m:%d::%H:%M:%y"))
+logging.basicConfig(filename=log_name, filemode="w",
+                    level=logging.DEBUG, format=formatter)
+logger = logging.getLogger("cfme_logger")
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.INFO)
+console_handler.setFormatter(logging.Formatter(formatter))
+logger.addHandler(console_handler)
+
+
+class CfmeUpgradeManeger(IPAppliance):
+
+    def __init__(self, appliance_ip, repo_list, dest_version):
+        super(CfmeUpgradeManeger, self).__init__(address=appliance_ip)
+        self.repo_list = repo_list
+        self.dest_version = dest_version
+
+    def exec_commaned(self, cmd, expect_failure=False, ignore_failure=False):
+        '''
+        execute command on the appliance and logging the command with it output to log file
+        :param cmd: command
+        :param expect_failure: is the command have to fail
+        :param ignore_failure: not raise exception on command failure
+        :return: command results
+        '''
+
+        logger.debug("exec_commaned: {cmd}".format(cmd=cmd))
+        result = self.ssh_client.run_command(cmd)
+        try:
+            if bool(result.rc) != expect_failure and not ignore_failure:
+                exception_msg = "command {cmd} failed!!\n\n\n" \
+                                "{command_output}".format(cmd=cmd,
+                                                          command_output=result.output)
+                raise RuntimeError(exception_msg)
+        except Exception as ex:
+            logger.exception(ex.message)
```

Comment:
logger.exception("It broke") will log the exception class and the formatted exception. So your code is producing doubled log lines which is no good.

```
In [5]: try: raise Exception("It was wind")
except Exception:
     logging.exception("It fell!")
   ...:     
ERROR:root:It fell!
Traceback (most recent call last):
  File "<ipython-input-5-8a089cb33723>", line 1, in <module>
    try: raise Exception("It was wind")
Exception: It was wind
```